figenv.py

```diff
@@ -75,6 +79,29 @@ def _to_float(cls, value):
     def _to_dict(cls, value):
         return json.loads(value)
 
+    def _to_list(cls, value):
+        return json.loads(value)
+
+    def _get_coerce_function(cls, value, annotation=None):
+        if not isinstance(value, str):
+            return None
+
+        if annotation is not None:
+            annoname = getattr(annotation, '_name', getattr(annotation, '__name__', None))
+            coerce_func = getattr(annotation, '_coerce', getattr(cls, f'_to_{annoname.lower()}', None))
+            # DEV: It may be more user friendly to raise an error here if we could not find a coercion method
+        else:
```

Comment:
This is the important change. Now the coerce_func guessing only happens if the configuration is not type annotated.

Reply:
I am down for this. If the object is type annotated, then we should be using the annotation to coerce only.